fingerpaintview/src/main/java/tech/picnic/fingerpaintview/FingerPaintImageView.kt

```diff
@@ -164,14 +188,54 @@ class FingerPaintImageView @JvmOverloads constructor(context: Context,
 
     override fun onDraw(canvas: Canvas?) {
         super.onDraw(canvas)
-        paths.forEach { canvas?.drawPath(it.first, it.second) }
+        mBitmap = Bitmap.createBitmap(widthBitmap,
+                heightBitmap,
+                Bitmap.Config.ARGB_8888)
+        mCanvas = Canvas(mBitmap)
+
+        canvas?.save()
+        for (index in paths.indices){
+            if(index >= countDrawn) {
+                if (isEmboos) paths[index].second.maskFilter = mEmboss
```

Comment:
`paths[index].second` is used on multiple places. It can be extracted into a separate variable for simplicity.